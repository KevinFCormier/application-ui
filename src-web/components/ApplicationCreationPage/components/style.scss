/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2020. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
@import 'node_modules/carbon-components/scss/globals/scss/_colors';
@import 'scss/variables';

.timeWindow-container {
  position: relative;

  .timeWindow-btn-group {
    display: inline-grid;
  }

  .timeWindow-btn {
    margin: 5px 0;
  }

  .timeWindow-config-days-checkboxes {
    width: 50%;
    display: grid;
    grid-template-columns: auto auto;

    .first-col {
      grid-column-start: 1;
    }

    .second-col {
      grid-column-start: 2;
    }
  }

  .timeWindow-config-timezone-dropdown {
    width: 180px;
  }

  .timeWindow-config-start-end-times {
    display: block;
  }

  .timeWindow-config-start {
    float: left;
  }

  .bx--select--inline {
    svg.bx--select__arrow {
      position: relative;
    }
  }
}