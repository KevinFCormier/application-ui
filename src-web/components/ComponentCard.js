/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import React from 'react'
import PropTypes from 'prop-types'
import { Icon } from 'carbon-components-react'
import {
  Module,
  ModuleHeader,
  ModuleBody,
  Tag
} from 'carbon-addons-cloud-react'
import resources from '../../lib/shared/resources'
import msgs from '../../nls/platform.properties'
import { Link } from 'react-router-dom'
import config from '../../lib/shared/config'

resources(() => {
  require('../../scss/dashboard-card.scss')
})

function formatStatus(statusArray, value) {
  const statuses = JSON.parse(statusArray).map(status => {
    return (
      <li
        className={`dashboard-card-text-list-item${value ? '' : '-alt'}`}
        key={status}
      >
        {status}
      </li>
    )
  })
  return <ul className="dashboard-card-text-list">{statuses}</ul>
}

const DashboardOrb = ({
  sev = 'healthy',
  status = ['healthy'],
  value,
  className = ''
}) => (
  <div className={`dashboard-count ${className}`}>
    <div className="dashboard-orb">
      {value === 0 ? (
        <div
          className={'dashboard-orb dashboard-orb__inner dashboard-orb__gray'}
        >
          {value}
        </div>
      ) : (
        <div
          className={`dashboard-orb dashboard-orb__inner dashboard-orb__${sev}`}
        >
          {value}
        </div>
      )}
    </div>
    {value === 0 ? (
      <div className={'dashboard-card-text dashboard-card-text__gray'}>
        {formatStatus(status, false)}
      </div>
    ) : (
      <div className={'dashboard-card-text'}>{formatStatus(status, true)}</div>
    )}
  </div>
)

const OrbPropType = {
  className: PropTypes.string,
  sev: PropTypes.string,
  status: PropTypes.string,
  value: PropTypes.number
}

DashboardOrb.propTypes = OrbPropType

const getIcon = status => {
  switch (status) {
  case 'healthy':
    return 'icon--checkmark--glyph'
  case 'warning':
    return 'icon--warning--glyph'
  case 'critical':
    return 'icon--error--glyph'
  default:
    break
  }
}

const getTableStatus = (critical, healthy, warning) => {
  switch (true) {
  case critical > 0:
    return 'critical'
  case warning > 0:
    return 'warning'
  default:
    return 'healthy'
  }
}

const getStatusSearchArray = (resourceName, status, locale) => {
  const statusList = {
    releases: {
      critical: [
        msgs.get('table.cell.failed', locale),
        msgs.get('table.cell.deleted', locale),
        msgs.get('table.cell.superseded', locale),
        msgs.get('table.cell.unknown', locale)
      ],
      warning: [msgs.get('table.cell.deleting', locale)],
      healthy: [msgs.get('table.cell.deployed', locale)]
    },
    pods: {
      critical: [
        msgs.get('table.cell.failed', locale),
        msgs.get('table.cell.notrunning', locale)
      ],
      warning: [msgs.get('table.cell.pending', locale)],
      healthy: [
        msgs.get('table.cell.running', locale),
        msgs.get('table.cell.succeeded', locale)
      ]
    },
    storage: {
      critical: [msgs.get('table.cell.unbound', locale)],
      warning: [msgs.get('table.cell.available', locale)],
      healthy: [msgs.get('table.cell.bound', locale)]
    },
    clusters: {
      critical: [msgs.get('table.cell.offline', locale)],
      healthy: [msgs.get('table.cell.ready', locale)]
    }
  }

  const tempStatusArray = statusList[resourceName][status] || [status]
  return JSON.stringify(tempStatusArray)
}

export class DashboardCard extends React.PureComponent {
  render() {
    const { locale } = this.context
    const {
      critical = 0,
      healthy = 0,
      title,
      warning = 0,
      type = ''
    } = this.props
    const cardStatus = getTableStatus(critical, healthy, warning)
    const totalCount = critical + warning + healthy
    return (
      <Module
        className={`bx--tile dashboard-card dashboard-card__${cardStatus}`}
        size="single"
        {...this.props}
      >
        <ModuleHeader className={`dashboard-card-header__${cardStatus}`}>
          <div className="dashboard-card-header__icon">
            <div className="dashboard-card-header__icon-title">{title}</div>
            <Tag type="beta">{totalCount}</Tag>
            {cardStatus !== 'healthy' ? (
              <Icon
                className={`dashboard-card-header__icon__${cardStatus}`}
                name={getIcon(cardStatus)}
                description={`table-status-icon-${cardStatus}`}
                role="img"
              />
            ) : null}
          </div>
        </ModuleHeader>
        <ModuleBody className={'dashboard-card-orb-summary'}>
          <div className="dashboard-overview">
            <div>
              <Link
                to={`${
                  config.contextPath
                }/${type}?filters={"status":${getStatusSearchArray(
                  type,
                  'critical',
                  locale
                )}}`}
              >
                <DashboardOrb
                  key="critical-orb"
                  sev="critical"
                  status={getStatusSearchArray(type, 'critical', locale)}
                  value={critical}
                />
              </Link>
              {type !== 'clusters' ? (
                <Link
                  to={`${
                    config.contextPath
                  }/${type}?filters={"status":${getStatusSearchArray(
                    type,
                    'warning',
                    locale
                  )}}`}
                >
                  <DashboardOrb
                    key="warning-orb"
                    sev="warning"
                    status={getStatusSearchArray(type, 'warning', locale)}
                    value={warning}
                  />
                </Link>
              ) : null}
              <Link
                to={`${
                  config.contextPath
                }/${type}?filters={"status":${getStatusSearchArray(
                  type,
                  'healthy',
                  locale
                )}}`}
              >
                <DashboardOrb
                  key="healthy-orb"
                  sev="healthy"
                  status={getStatusSearchArray(type, 'healthy', locale)}
                  value={healthy}
                />
              </Link>
            </div>
          </div>
        </ModuleBody>
        <ModuleBody className={'dashboard-card-table-body'}>
          <div className={'dashboard-card-separator-link'}>
            {type ? (
              <Link to={`${config.contextPath}/${type}`}>
                {msgs.get('dashboard.module.separator.link')}
              </Link>
            ) : null}
          </div>
        </ModuleBody>
      </Module>
    )
  }
}

DashboardCard.propTypes = {
  critical: PropTypes.number,
  healthy: PropTypes.number,
  namespace: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  warning: PropTypes.number
}
