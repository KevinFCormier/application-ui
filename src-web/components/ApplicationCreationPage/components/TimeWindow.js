/*******************************************************************************
 * Licensed Materials - Property of IBM
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import resources from '../../../../lib/shared/resources'
import {
  Accordion,
  AccordionItem,
  Checkbox,
  DropdownV2,
  Icon,
  RadioButton,
  RadioButtonGroup,
  SelectItem,
  TimePicker,
  TimePickerSelect
} from 'carbon-components-react'
import Tooltip from '../../TemplateEditor/components/Tooltip'
import msgs from '../../../../nls/platform.properties'

resources(() => {
  require('./style.scss')
})

class TimeWindow extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    handleChange: PropTypes.func,
    locale: PropTypes.string
  };

  constructor(props) {
    super(props)
    this.state = {}
    this.props.control.active = { mode: '', days: [], timezone: '' }
  }

  render() {
    const { locale, control } = this.props
    const { name, validation = {} } = control
    const modeSelected = control.active.mode ? true : false
    const daysSelectorID = 'days-selector'
    const timezoneDropdownID = 'timezone-dropdown'

    return (
      <React.Fragment>
        <div className="creation-view-controls-labels">
          <div className="creation-view-controls-textarea-title">
            {name}
            {validation.required ? (
              <div className="creation-view-controls-required">*</div>
            ) : null}
            <Tooltip control={control} locale={locale} />
          </div>

          <div className="timeWindow-container">
            <RadioButtonGroup
              className="timeWindow-mode-container"
              name="timeWindow-mode-container"
            >
              <RadioButton
                className="mode-btn"
                id="active-mode"
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.activeMode',
                  locale
                )}
                value="&quot;active&quot;"
                onClick={this.handleChange.bind(this)}
              />
              <RadioButton
                className="mode-btn"
                id="block-mode"
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.blockMode',
                  locale
                )}
                value="&quot;block&quot;"
                onClick={this.handleChange.bind(this)}
              />
            </RadioButtonGroup>

            <Accordion>
              <AccordionItem
                open={modeSelected}
                title={msgs.get(
                  'creation.app.settings.timeWindow.config',
                  locale
                )}
              >
                <div className="timeWindow-config-container">
                  <div className="config-days-section">
                    <div className="config-title">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.days.title',
                        locale
                      )}{' '}
                      *
                    </div>
                    <div className="config-descr">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.days.descr',
                        locale
                      )}
                    </div>
                    <div className="config-days-selector">
                      <div className="first-col">
                        <Checkbox
                          labelText="Monday"
                          name={daysSelectorID}
                          id="mon"
                          value="&quot;Monday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Tuesday"
                          name={daysSelectorID}
                          id="tues"
                          value="&quot;Tuesday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Wednesday"
                          name={daysSelectorID}
                          id="wed"
                          value="&quot;Wednesday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Thursday"
                          name={daysSelectorID}
                          id="thurs"
                          value="&quot;Thursday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Friday"
                          name={daysSelectorID}
                          id="fri"
                          value="&quot;Friday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                      </div>
                      <div className="second-col">
                        <Checkbox
                          labelText="Saturday"
                          name={daysSelectorID}
                          id="sat"
                          value="&quot;Saturday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Sunday"
                          name={daysSelectorID}
                          id="sun"
                          value="&quot;Sunday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                      </div>
                    </div>
                  </div>

                  <div className="config-timezone-section">
                    <div className="config-title">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.timezone.title',
                        locale
                      )}{' '}
                      *
                    </div>
                    <DropdownV2
                      className="config-timezone-dropdown"
                      label="Choose a location"
                      items={[
                        {
                          label: 'America/Edmonton',
                          name: timezoneDropdownID,
                          value: '"America/Edmonton"'
                        },
                        {
                          label: 'America/Halifax',
                          name: timezoneDropdownID,
                          value: '"America/Halifax"'
                        },
                        {
                          label: 'America/Toronto',
                          name: timezoneDropdownID,
                          value: '"America/Toronto"'
                        },
                        {
                          label: 'America/Vancouver',
                          name: timezoneDropdownID,
                          value: '"America/Vancouver"'
                        },
                        {
                          label: 'America/Winnipeg',
                          name: timezoneDropdownID,
                          value: '"America/Winnipeg"'
                        }
                      ]}
                      disabled={!modeSelected}
                      onChange={this.handleChange.bind(this)}
                    />
                  </div>

                  <div className="config-time-section">
                    <div className="config-start-time">
                      <TimePicker
                        id="start-time"
                        labelText="Start Time"
                        maxLength={5}
                        placeholder="hh:mm"
                        type="text"
                        disabled={!modeSelected}
                      >
                        <TimePickerSelect
                          id="start-time-selector"
                          labelText="AM/PM"
                          placeholder="hh:mm"
                          disabled={!modeSelected}
                        >
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="AM"
                            value="AM"
                          />
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="PM"
                            value="PM"
                          />
                        </TimePickerSelect>
                      </TimePicker>
                    </div>
                    <div className="config-end-time">
                      <TimePicker
                        id="end-time"
                        labelText="End Time"
                        maxLength={5}
                        placeholder="hh:mm"
                        type="text"
                        disabled={!modeSelected}
                      >
                        <TimePickerSelect
                          id="end-time-selector"
                          labelText="AM/PM"
                          placeholder="hh:mm"
                          disabled={!modeSelected}
                        >
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="AM"
                            value="AM"
                          />
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="PM"
                            value="PM"
                          />
                        </TimePickerSelect>
                      </TimePicker>
                    </div>
                    <div className="add-time-btn">
                      <Icon
                        name="icon--add--glyph"
                        fill="#3d70b2"
                        description=""
                        className="add-time-btn-icon"
                      />
                      <div className="add-time-btn-text">
                        {msgs.get(
                          'creation.app.settings.timeWindow.config.time.add',
                          locale
                        )}
                      </div>
                    </div>
                  </div>
                </div>
              </AccordionItem>
            </Accordion>

            {/* <TextArea
              id={id}
              name="timeWindow-text"
              invalid={!!exception}
              invalidText={exception}
              hideLabel
              spellCheck={false}
              autoComplete={'no'}
              labelText=""
              onChange={this.handleChange.bind(this)}
            /> */}
          </div>
        </div>
      </React.Fragment>
    )
  }

  handleChange(event) {
    const { control, handleChange } = this.props
    let targetName = ''
    try {
      targetName = event.target.name
    } catch (e) {
      targetName = ''
    }

    if (targetName && targetName === 'timeWindow-mode-container') {
      control.active.mode = event.target.value
    } else if (targetName && targetName === 'days-selector') {
      if (event.target.checked === true) {
        control.active.days.push(event.target.value)
      } else {
        const index = control.active.days.indexOf(event.target.value)
        control.active.days.splice(index, 1)
      }
    } else if (
      event.selectedItem &&
      event.selectedItem.name === 'timezone-dropdown'
    ) {
      control.active.timezone = event.selectedItem.value
    }

    // else if (event.target.name === 'timeWindow-text') {
    //   control.active.text = '"' + event.target.value + '"'
    // }
    handleChange(control)
  }
}

export default TimeWindow
