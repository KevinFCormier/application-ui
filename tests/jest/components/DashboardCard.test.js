/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import { DashboardCard } from '../../../src-web/components/ComponentCard'

describe('DashboardCard component 1', () => {
  const table = [
    {
      'status': 'critical',
      'clusterIP': '1.1.1.1',
      'resourceName': 'auth-idp-platform-auth-cert-gen-h2rt4',
      'percentage': 0
    },
    {
      'status': 'warning',
      'clusterIP': '',
      'resourceName': 'db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz',
      'percentage': 0
    },
    {
      'status': 'healthy',
      'clusterIP': '',
      'resourceName': 'db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz',
      'percentage': 0
    }
  ]
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <DashboardCard critical={1} healthy={1} warning={1} title='unit test' table={table} type='pods' />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})


describe('DashboardCard component 2', () => {
  const table = [
    {
      'status': 'warning',
      'clusterIP': '',
      'resourceName': 'db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz',
      'percentage': 0
    }
  ]
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <DashboardCard critical={0} healthy={0} warning={1} title='unit test 2' table={table} type='pods' />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})


describe('DashboardCard component 3', () => {
  const table = [
    {
      'clusterIP': '',
      'resourceName': 'db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz',
      'percentage': 0
    }
  ]
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <DashboardCard critical={0} healthy={1} warning={0} title='unit test 3' table={table} type='pods' />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('DashboardCard component 4', () => {
  const table = []
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <DashboardCard critical={0} healthy={0} warning={0} title='unit test 4' table={table} type='pods' />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
