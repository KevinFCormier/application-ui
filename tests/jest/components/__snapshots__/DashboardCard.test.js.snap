// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCard component 1 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card dashboard-card__critical"
  critical={1}
  healthy={1}
  table={
    Array [
      Object {
        "clusterIP": "1.1.1.1",
        "percentage": 0,
        "resourceName": "auth-idp-platform-auth-cert-gen-h2rt4",
        "status": "critical",
      },
      Object {
        "clusterIP": "",
        "percentage": 0,
        "resourceName": "db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz",
        "status": "warning",
      },
      Object {
        "clusterIP": "",
        "percentage": 0,
        "resourceName": "db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz",
        "status": "healthy",
      },
    ]
  }
  title="unit test"
  type="pods"
  warning={1}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__critical"
    >
      <h1
        className="bx--module__title"
      >
        <div
          className="dashboard-card-header__icon"
        >
          unit test
          <svg
            alt="table-status-icon-critical"
            aria-label="table-status-icon-critical"
            className="dashboard-card-header__icon__critical"
            fill={undefined}
            fillRule="evenodd"
            height="16"
            name="icon--error--glyph"
            role="img"
            style={undefined}
            viewBox="0 0 16 16"
            width="16"
          >
            <title>
              table-status-icon-critical
            </title>
            <path
              d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM5.1 13.3L3.5 12 11 2.6l1.5 1.2-7.4 9.5z"
            />
          </svg>
        </div>
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb-summary"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Failed\\",\\"Not running\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__critical"
                >
                  1
                </div>
              </div>
              <div
                className="dashboard-card-text"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Failed
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Not running
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Pending\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__warning"
                >
                  1
                </div>
              </div>
              <div
                className="dashboard-card-text"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Pending
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Running\\",\\"Succeeded\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__healthy"
                >
                  1
                </div>
              </div>
              <div
                className="dashboard-card-text"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Running
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Succeeded
                  </li>
                </ul>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/pods"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardCard component 2 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card dashboard-card__warning"
  critical={0}
  healthy={0}
  table={
    Array [
      Object {
        "clusterIP": "",
        "percentage": 0,
        "resourceName": "db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz",
        "status": "warning",
      },
    ]
  }
  title="unit test 2"
  type="pods"
  warning={1}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__warning"
    >
      <h1
        className="bx--module__title"
      >
        <div
          className="dashboard-card-header__icon"
        >
          unit test 2
          <svg
            alt="table-status-icon-warning"
            aria-label="table-status-icon-warning"
            className="dashboard-card-header__icon__warning"
            fill={undefined}
            fillRule="evenodd"
            height="16"
            name="icon--warning--glyph"
            role="img"
            style={undefined}
            viewBox="0 0 16 16"
            width="16"
          >
            <title>
              table-status-icon-warning
            </title>
            <path
              d="M8 1L0 15h16L8 1zm-.8 5h1.5v1.4L8.3 11h-.8l-.4-3.6V6h.1zm.8 8c-.6 0-1-.4-1-1s.4-1 1-1 1 .4 1 1-.4 1-1 1z"
            />
          </svg>
        </div>
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb-summary"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Failed\\",\\"Not running\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Failed
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Not running
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Pending\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__warning"
                >
                  1
                </div>
              </div>
              <div
                className="dashboard-card-text"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Pending
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Running\\",\\"Succeeded\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Running
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Succeeded
                  </li>
                </ul>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/pods"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardCard component 3 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card dashboard-card__healthy"
  critical={0}
  healthy={1}
  table={
    Array [
      Object {
        "clusterIP": "",
        "percentage": 0,
        "resourceName": "db2-stock-trader-ibm-db2oltp-dev-5b77c44d4-b7tbz",
      },
    ]
  }
  title="unit test 3"
  type="pods"
  warning={0}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__healthy"
    >
      <h1
        className="bx--module__title"
      >
        <div
          className="dashboard-card-header__icon"
        >
          unit test 3
        </div>
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb-summary"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Failed\\",\\"Not running\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Failed
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Not running
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Pending\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Pending
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Running\\",\\"Succeeded\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__healthy"
                >
                  1
                </div>
              </div>
              <div
                className="dashboard-card-text"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Running
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Succeeded
                  </li>
                </ul>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/pods"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardCard component 4 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card dashboard-card__healthy"
  critical={0}
  healthy={0}
  table={Array []}
  title="unit test 4"
  type="pods"
  warning={0}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__healthy"
    >
      <h1
        className="bx--module__title"
      >
        <div
          className="dashboard-card-header__icon"
        >
          unit test 4
        </div>
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb-summary"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Failed\\",\\"Not running\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Failed
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Not running
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Pending\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Pending
                  </li>
                </ul>
              </div>
            </div>
          </a>
          <a
            href="undefined/pods?filters={\\"status\\":[\\"Running\\",\\"Succeeded\\"]}"
            onClick={[Function]}
          >
            <div
              className="dashboard-count "
            >
              <div
                className="dashboard-orb"
              >
                <div
                  className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
                >
                  0
                </div>
              </div>
              <div
                className="dashboard-card-text dashboard-card-text__gray"
              >
                <ul
                  className="dashboard-card-text-list"
                >
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Running
                  </li>
                  <li
                    className="dashboard-card-text-list-item"
                  >
                    Succeeded
                  </li>
                </ul>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/pods"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;
